app/src/main/java/com/example/habitsmasher/ui/dashboard/AddHabitDialog.java

```diff
@@ -80,15 +96,112 @@ public void onClick(View view) {
+                if (tracker.getMonday()){
+                    tracker.setMonday(false);
+                }
+                //if monday wasn't already selected, select it
+                else{
+                    tracker.setMonday(true);
+                }
+                Log.d("Tracker Status", tracker.getDays());
+            }
+        });
+
+        tuesdayButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                if(tracker.getTuesday()){
+                    tracker.setTuesday(false);
+                }
+                else{
+                    tracker.setTuesday(true);
+                }
+                Log.d("Tracker Status", tracker.getDays());
+            }
+        });
+
+        wednesdayButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                if(tracker.getWednesday()){
+                    tracker.setWednesday(false);
+                }
+                else{
+                    tracker.setWednesday(true);
+                }
+                Log.d("Tracker Status", tracker.getDays());
+            }
+        });
+
+        thursdayButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                if(tracker.getThursday()){
+                    tracker.setThursday(false);
+                }
+                else{
+                    tracker.setThursday(true);
+                }
+                Log.d("Tracker Status", tracker.getDays());
+            }
+        });
+
+        fridayButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                if(tracker.getFriday()){
+                    tracker.setFriday(false);
+                }
+                else{
+                    tracker.setFriday(true);
+                }
+                Log.d("Tracker Status", tracker.getDays());
+            }
+        });
+
+        saturdayButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                if(tracker.getSaturday()){
+                    tracker.setSaturday(false);
+                }
+                else{
+                    tracker.setSaturday(true);
+                }
+                Log.d("Tracker Status", tracker.getDays());
+            }
+        });
+
+        sundayButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                if(tracker.getSunday()){
+                    tracker.setSunday(false);
+                }
+                else{
+                    tracker.setSunday(true);
+                }
+                Log.d("Tracker Status", tracker.getDays());
+            }
+        });
```

Comment:
Can we extract this huge chunk of code to a helper method called `setClickListenersForDaysOfTheWeek()`, that way the parent method is more readable + concise.

Reply:
Extracted, in addition to making the buttons and the tracker private.